### ISSUE TITLE
Cannot place a variable in include statement

### ISSUE THREAD

**Issue by suratdas** [unclebob/fitnesse#879]
I have test wiki page in one area which is being included from another area. Now the root of those page (Say RegressionRoot ) may change when we decide to rename a parent folder or move the wiki pages from Windows Explorer instead of doing Tools->Move. We want to do this from Windows Explorer because it is very fast and can be scripted (e.g. from Jenkins). 

I plan to define a variable in Fitnesse root and refer this variable in the tests to evaluate the root of regression test pages. However, the variable is not taken in "!include" statement as can be seen in the screenshot below. I have included manually in the last line in the screenshot to show that the path is a valid path. If this is not supported, any plan to include this feature in upcoming releases?

![image](https://cloud.githubusercontent.com/assets/9042580/13298543/fa2b4370-daeb-11e5-9c72-0a9f45d6afff.png)

**Comment by jediwhale** [comment]
You need ${RegressionRoot} - it looks like you have just {RegressionRoot}

On 2016-02-24 12:54, Surat Das wrote:
>
> I have test wiki page in one area which is being included from another 
> area. Now the root of those page (Say RegressionRoot ) may change when 
> we decide to rename a parent folder or move the wiki pages from 
> Windows Explorer instead of doing Tools->Move. We want to do this from 
> Windows Explorer because it is very fast and can be scripted (e.g. 
> from Jenkins).
>
> I plan to define a variable in Fitnesse root and refer this variable 
> in the tests to evaluate the root of regression test pages. However, 
> the variable is not taken in "!include" statement as can be seen in 
> the screenshot below. I have included manually in the last line in the 
> screenshot to show that the path is a valid path. If this is not 
> supported, any plan to include this feature in upcoming releases?
>
> image 
> <https://cloud.githubusercontent.com/assets/9042580/13298543/fa2b4370-daeb-11e5-9c72-0a9f45d6afff.png>
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/unclebob/fitnesse/issues/879>.
>

-- 
Cheers,
Mike Stockdale

/fit/Sharp <http://fitsharp.github.com>

**Comment by suratdas** [comment]
Thanks for pointing. My mistake, I corrected that one but it still does not resolve to the whole path. Please see this screenshot.
![image](https://cloud.githubusercontent.com/assets/9042580/13301219/bba78444-daf8-11e5-9d51-a891caa7f299.png)

**Comment by suratdas** [comment]
My mistake. I corrected that but it still does not resolve to the whole path. Please see this screenshot.
![image](https://cloud.githubusercontent.com/assets/9042580/13301261/f2cfb928-daf8-11e5-8e47-b56d44d86ff9.png)

**Comment by suratdas** [comment]
On further checking I figured out that even if I give the whole path in the variable, the variable is recognized but the page is not included. Please see this screenshot.

![image](https://cloud.githubusercontent.com/assets/9042580/13307004/0f112b5c-db1d-11e5-95aa-2572fc899734.png)

**Comment by jediwhale** [comment]
Yes, you're right.  Fitnesse does not support variables in an include, 
so this is a request for an enhancement.

On 2016-02-24 18:36, Surat Das wrote:
>
> On further checking I figured out that even if I give the whole path 
> in the variable, the variable is recognized but the page is not 
> included. Please see this screenshot.
>
> image 
> <https://cloud.githubusercontent.com/assets/9042580/13307004/0f112b5c-db1d-11e5-95aa-2572fc899734.png>
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/unclebob/fitnesse/issues/879#issuecomment-188552019>.
>

-- 
Cheers,
Mike Stockdale

/fit/Sharp <http://fitsharp.github.com>

**Comment by six42** [comment]
Hi @suratdas 

Not exactly what you want but the below works. Let us know it it helps.

!define INC_ALL {!include .FitNesse.UserGuide.FitNesseWiki.MarkupLanguageReference.MarkupVariables}

${INC_ALL}

**Comment by the-noob** [comment]
+1 this, would be very useful when using different test platforms like 'dev' and 'staging' and you want to include some credentials. I.e.

.MySuite
!define PLATFORM {Test}

.MySuiteTest
!include .Credentials.${PLATFORM}
